docs/docs/understanding/workflows/resources.md

```diff
@@ -1,12 +1,16 @@
 # Resources
 
-Resources are a component of workflows that allow us to equip our steps with external dependencies such as memory, LLMs, query engines or chat history.
+Resources are external dependencies such as memory, LLMs, query engines or chat history instances that will be injected
+into workflow steps at runtime.
 
-Resources are a powerful way of binding components to our steps that we otherwise would need to specify by hand every time and, most importantly, resources are **stateful**, meaning that they maintain their state across different steps, unless otherwise specified.
+Resources are a powerful way of binding workflow steps to Python objects that we otherwise would need to create by hand
+every time. For performance reasons, by default resources are cached for a workflow, meaning the same resource instance
+is passed to every step where it's injected. It's important to master this concept because cached and non-cached
+resources can lead to unexpected behaviour, let's see it in detail.
 
-## Using Stateful Resources
```

Comment:
This is the biggest change in this PR. I don't believe we can link statefulness to caching behaviour, addded a paragraph explaining their relationship at the end